### PR TITLE
MEN-3617: form to upgrade the tenant

### PR DESCRIPTION
**Pull request by tranchitella**

### CODE REVIEW

File: src/js/components/settings/cardsection.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
+/**
+* Use the CSS tab above to style your Element's container.
+*/
```
Comment (mzedel) [inline src/js/components/settings/cardsection.js:3]: that is probably a remainder from the stripe website...

File: src/js/components/settings/settings.js [hunk 2]
```diff
@@ -76,8 +85,10 @@ const mapStateToProps = state => {
     allowUserManagement,
     currentUser,
     isAdmin,
+    isHosted: state.app.features.isHosted,
     isEnterprise: state.app.features.isEnterprise || (state.app.features.isHosted && plan === 'enterprise'),
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply src/js/components/settings/settings.js:89]: There is no trial plan, trial is a boolean attribute in the tenant. A trial account has trial = true, plan = 'enterprise'.
Why is this breaking the mapping?

File: src/js/components/settings/upgrade.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      {
+        name: 'Starter', price: '$29/month for first 50 devices', features: 'Basic features', value: 'os'
+      },
+      {
+        name: 'Professional', price: '$249/month for first 250 devices', features: '+ Extra features', value: 'professional'
+      },
+      {
+        name: 'Enterprise', price: 'Custom pricing, unlimited devices', features: '+ Full features + SLA support', value: 'enterprise'
+      },
+    ];
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply src/js/components/settings/upgrade.js:53]: Moved outside the class.

File: src/js/components/settings/cardsection.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        // The card setup has succeeded. Display a success message and send
+        // to our backend
+        try {
+          props.completeUpgrade(props.org.id, props.plan).then(() => {
+            setLoading(false);
+            props.upgradeSuccess();
+          });
+        } catch (err) {
+          props.setSnackbar(preformatWithRequestID(err.res, `There was an error upgrading your account. ${err.res.body.error}`));
+        }
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply src/js/components/settings/cardsection.js:68]: There is no need to call cancelUpgrade in the backend if completeUpgrade fails; you are right regarding the action promise chain, fxied.

File: src/js/components/settings/upgrade.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+      disabled: false,
+      updatedFormdata: {plan: 'os', name: 'Starter'},
+    };
+  }
+  componentDidMount() {
+    this.props.getUserOrganization();
+  }
+
+  changePlan(plan, name) {
+    const updatedFormdata = { ...this.state.updatedFormdata, plan: plan, name: name };
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply src/js/components/settings/upgrade.js:23]: Done

File: src/js/components/settings/cardsection.js (hunk 1 above)
Comment (tranchitella) [reply src/js/components/settings/cardsection.js:3]: yup

File: src/js/components/settings/settings.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (mzedel) [reply src/js/components/settings/settings.js:89]: I should have marked the related part as well: https://github.com/mendersoftware/gui/pull/798/files#diff-feb05faa1e057f773084ca8397908781R70 there the plan is set to `trial` if the trial attribute is set, thus the check for `isEnterprise`/ the enterprise plan fails. this then leads to the roles config being unavailable as they are checked to be present only for enterprise admins

[the remark this answers is not in the document]
Comment (tranchitella) [reply src/js/components/settings/settings.js:89]: To be honest, I've tested it and I could see the roles in the UI.
I'll try it again later on (I'm on a back-end task now).

[the remark this answers is not in the document]
Comment (mzedel) [reply src/js/components/settings/settings.js:89]: you will see the roles if you are running onprem enterprise  - but not if you are running hosted with an enterprise plan

[the remark this answers is not in the document]
Comment (tranchitella) [reply src/js/components/settings/settings.js:90]: ok, got it and sorry for the noise, you are (obviously) right.
I've fixed it in the right way, using the `organization.trial` boolean property.

[the remark this answers is not in the document]
Comment (mzedel) [reply src/js/components/settings/settings.js:90]: Perfect! thank you!

**Pull request merged**